### PR TITLE
[13.0][FW-PORT] mrp_multi_level: Manage Kits in MRP Multi Level

### PR DESCRIPTION
**Pull request by BernatPForgeFlow**
Forward port of https://github.com/OCA/manufacture/pull/842.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@BernatPForgeFlow I think that you messed up https://github.com/OCA/manufacture/pull/848/files#diff-71f99ae0d3cbd0109e88e3dea79d6d97292f7fb4509e44dd54599b732b81ddaaR41

File: mrp_multi_level/wizards/mrp_multi_level.py [hunk 1]
```diff
@@ -758,8 +759,11 @@ def _mrp_final_process(self, mrp_areas):
 
         for product_mrp_area in product_mrp_area_ids:
             # Build the time-phased inventory
-            if self._exclude_from_mrp(
-                product_mrp_area.product_id, product_mrp_area.mrp_area_id
+            if (
+                self._exclude_from_mrp(
+                    product_mrp_area.product_id, product_mrp_area.mrp_area_id
+                )
+                or product_mrp_area.supply_method == "phantom"
```
Comment (LoisRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py:784]: This condition should be included as the default condition in the `_exclude_from_mrp`method.

File: mrp_multi_level/wizards/mrp_multi_level.py [hunk 2]
```diff
@@ -502,13 +503,13 @@ def _init_mrp_move_grouped_demand(self, nbr_create, product_mrp_area):
...
                 )
             ):
+                origin = ",".join(list(set(demand_origin)))
                 name = _(
                     "Grouped Demand of %(product_name)s for %(delta_days)d Days"
                 ) % dict(
                     product_name=product_mrp_area.product_id.display_name,
                     delta_days=grouping_delta,
                 )
-                origin = ",".join(list(set(demand_origin)))
```
Comment (LoisRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py]: why this change?

File: mrp_multi_level/wizards/mrp_multi_level.py [hunk 3]
```diff
@@ -321,9 +324,7 @@ def _low_level_code_calculation(self):
             products = bom_lines.mapped("product_id")
             products.write({"llc": llc})
             counter = self.env["product.product"].search_count([("llc", "=", llc)])
-            log_msg = "Low level code {} finished - Nbr. products: {}".format(
-                llc, counter
-            )
+            log_msg = "Low level code %s finished - Nbr. products: %s" % (llc, counter)
```
Comment (LoisRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py]: and this one?

File: mrp_multi_level/wizards/mrp_multi_level.py [hunk 4]
```diff
@@ -157,7 +157,7 @@ def _prepare_mrp_move_data_bom_explosion(
             ).replace(
                 "Demand Bom Explosion: Demand Bom Explosion: ", "Demand Bom Explosion: "
             ),
-            "origin": planned_order.origin,
+            "origin": planned_order.origin if planned_order else name[-12:],
```
Comment (LoisRForgeFlow) [inline mrp_multi_level/wizards/mrp_multi_level.py]: this change?

**Review by LoisRForgeFlow — APPROVED** [review]
Looks great now, thanks!

/ocabot merge minor
